samwise.js

```diff
@@ -1,34 +1,37 @@
 /**
```

Comment:
SCRIPT-This block of code could benefit from some comments explaining the logic behind it.